### PR TITLE
Feature/quantity

### PR DESCRIPTION
**Pull request by Jammy2211**
This initially set out as a feature which allows one to directly fit a quantity of a plane or tracer (e.g. the deflection angle map) with a lens model, omitted the need to fit a model with data.

In order to plot the results of such a fit, this PR then turned into a large refactor of the plot modules including thorough documentation.

### CODE REVIEW

File: autoarray/plot/mat_wrap/mat_plot.py [hunk 1]
```diff
@@ -319,7 +319,7 @@ def plot_yx(
         y: Union[np.ndarray, List, Array1D],
         visuals_1d: Visuals1D,
         auto_labels: AutoLabels,
-        x: Optional[Union[np.ndarray, List, Array1D]] = None,
+        x: Optional[Union[np.ndarray, Iterable, List, Array1D]] = None,
```
Comment (rhayes777) [inline autoarray/plot/mat_wrap/mat_plot.py:322]: Would one of the types in this Union comprise all of the functionality expected of x?

File: test_autoarray/fit/plot/test_fit_imaging_plotters.py [hunk 2]
```diff
@@ -27,27 +27,48 @@ def test__visuals_in_constructor_use_imaging_and_include(fit_imaging_7x7):
     )
 
     assert fit_imaging_plotter.visuals_2d.origin == (1.0, 1.0)
-    assert fit_imaging_plotter.visuals_with_include_2d.origin == (1.0, 1.0)
+    assert fit_imaging_plotter.get_2d.via_fit_imaging_from(
```
Comment (rhayes777) [inline test_autoarray/fit/plot/test_fit_imaging_plotters.py:30]: Create fit_image_plotter in a fixture and separate out groups of assertions into different tests

**Review by rhayes777 — APPROVED** [review]

File: autoarray/plot/mat_wrap/mat_plot.py (hunk 1 above)
Comment (Jammy2211) [reply autoarray/plot/mat_wrap/mat_plot.py:322]: Possibly, but one still gets lint errors if you use for example an Iterable when you only type using np,ndarray

File: test_autoarray/fit/plot/test_fit_imaging_plotters.py (hunk 2 above)
Comment (Jammy2211) [reply test_autoarray/fit/plot/test_fit_imaging_plotters.py:30]: I actually meant to delete this unit test anyway.

File: autoarray/plot/mat_wrap/mat_plot.py (hunk 1 above)
Comment (rhayes777) [reply autoarray/plot/mat_wrap/mat_plot.py:322]: np.ndarray implements iterable so if you use the Iterable type that might be sufficient. I'm not actually looking at the code but it depends on how you use the variable; if you just iterate through x then it being of type Iterable would suffice.

There's a rule where you should make your argument types as general as possible and your return types as specific as possible. This makes composition of functions more flexible.

**Pull request merged**